### PR TITLE
Allows BAM with csi index to be used

### PR DESCRIPTION
**Pull request by keiranmraine**
Provide csi support for use of BAMs with csi already generated as discussed in #57.

*  Errors if csi is older than BAM indicating it can't handle generation of csi currently.
* Test added to confirm access via index works

I tested with HTSlib 1.7

### CODE REVIEW

File: lib/Bio/DB/HTS.pm [hunk 1]
```diff
@@ -2143,20 +2143,24 @@ sub index {
...
         elsif ( -e "${path}.crai" && mtime($path) > mtime("${path}.crai") ) {
             $self->reindex($path);
         }
     }
 
     croak "No index file for $path; try opening file with -autoindex"
       unless -e "${path}.bai" or
-      -e "${path}.crai";
+        -e "${path}.csi" or
+        -e "${path}.crai";
```
Comment (avullo) [inline lib/Bio/DB/HTS.pm:2163]: Aren't you actually suggesting to fail in the csi case by telling to rerun with autoindex?

File: t/01bam.t [hunk 2]
```diff
@@ -514,6 +514,15 @@ for my $use_fasta ( 0, 1 ) {
     ok( $matches{matched}/$matches{total} > 0.99 );
 } ## end for my $use_fasta ( 0, ...)
 
+{ # test access via csi index
```
Comment (avullo) [inline t/01bam.t:517]: This is fine, but for complete coverage, I would do something similar to the low/high level tests performed on ex1 to check whether certain operations with the csi index work as expected.

File: lib/Bio/DB/HTS.pm (hunk 1 above)
Comment (keiranmraine) [reply lib/Bio/DB/HTS.pm:2163]: I'm assuming at some point `-autoindex` will allow csi.

File: t/01bam.t (hunk 2 above)
Comment (keiranmraine) [reply t/01bam.t:517]: I can look into extending the tests if it's the only way to get this moving.  I only created the branch as we were getting no response or idea when something would happen.

File: lib/Bio/DB/HTS.pm (hunk 1 above)
Comment (avullo) [reply lib/Bio/DB/HTS.pm:2163]: Fair enough.

File: t/01bam.t (hunk 2 above)
Comment (avullo) [reply t/01bam.t:517]: I've made the comment, but haven't actually requested the mandatory change. Sorry if you feel communication isn't what you expect. Rishi has just recently handed over maintenance to the e! core team (specifically me). I'm gradually getting my way through this, but this task is not the priority of my job at the moment.

Comment (keiranmraine) [reply t/01bam.t:517]: I'm currently adding more robust tests by changing the ex1 tests to be reused for the ex3.bam with csi index.  Looking at this test class looks like I've written most of them over the years.  I'll raise a new pr once I've finished them

Comment (avullo) [reply t/01bam.t:517]: Most appreciated, thanks :+1:

**Pull request merged**